### PR TITLE
Add bind argument to delayed, to allow dependencies not implied by wrapped-function's args

### PR DESCRIPTION
**Pull request by aa1371**
- [x] Closes #7901
- [x] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask` / `isort dask`

Currently to apply deps not implied by function args in dask.delayed you can do this (thanks @abergou for the tip):

```
from dask.graph_manipulation import bind

t1 = delayed(task1)()
t2 = bind(delayed(task2), t1)()
```

While this works as expected, the invocation is a bit dense/hard to read, requires an extra import, and is not immediately obvious to users. What I'm proposing is that we provide a `bind` parameter to delayed that will then internally apply the bind to the delayed object. So the invocation would look like so:

```
t1 = delayed(task1)()
t2 = delayed(task2, bind=t1)()
```

This also has the added benefit of documenting this ability on the delayed interface, rather than being hidden where most users wouldn't know where to find it.

### CODE REVIEW

File: dask/delayed.py [hunk 1]
```diff
@@ -721,3 +739,9 @@ def __call__(self, *args, **kwargs):
 def single_key(seq):
     """Pick out the only element of this list, a list of keys"""
     return seq[0]
+
+
+# TODO: Need to reorganize modules to avoid circular import, and allow
+#       import of this at the top. As of now this import needs to come
+#       after the definition of `def delayed` and `class Delayed`.
+from .graph_manipulation import bind as _bind
```
Comment (aa1371) [inline dask/delayed.py:747]: This is obviously not ideal. Didn't want to spend to reorganizing if there is no appetite for this change. If there is, can look into this

**Comment by jsignell** [comment]
Thanks for writing this up @aa1371 and sorry it has taken some time for a review. To me, this seems like a rather large change for a slight simplification of a niche usecase. So I am not persuaded that it is necessary. I do recommend that you add an example to the documentation though!

**Review by jrbourbeau — COMMENTED** [review]
Yeah, I agree with @jsignell. This strikes me as a valid, though non-common use case. Instead of exposing `bind=` through the delayed interface, I'd rather guide users towards the API documented here https://docs.dask.org/en/latest/graph_manipulation.html which will work with all Dask collections (included delayed)

**Comment by aa1371** [comment]
Yea I was actually thinking after I opened this that just a quick mention in the delayed docs is probably good enough for this too. Will close this PR and open a new one with just a doc change soon. Thanks.

**Pull request closed without merging**